qiskit_experiments/curve_analysis/curve_data.py

```diff
@@ -126,6 +128,181 @@ def get_subset_of(self, index: Union[str, int]) -> "CurveData":
         )
 
 
+@dataclasses.dataclass(frozen=True)
+class CurveFitResult:
```

Comment:
This class has a lot of extra methods on top of a dataclass, maybe it should just be a regular class?